src/routes/guides/testing-your-app.mdx

```diff
@@ -0,0 +1,404 @@
+2. If it should be tested whether something is *not* rendered, first find something that will be rendered at the same time and then use `query...` to test if the result is null or an empty array.
+
+The last part depends on the characteristics of the element you want to select. If possible, try to select for accessible attributes (role, label, placeholder, text, ...):
+
+- **Role**: WAI ARIA landmark roles which are automatically set by semantic elements like `<button>` or otherwise with the `role`-attribute
+- **LabelText**: elements described by a label wrapping the element or is linked with `for`- or `aria-label`-attribute
+- **PlaceholderText**: input elements with a `placeholder`-attribute
+- **Text**: searches the text in all text nodes within the element, even if split over multiple nodes
+- **DisplayValue**: form elements showing the given value
+- **AltText**: images with the alt text
+- **Title**: HTML elements with the `title`-attribute or SVGs with the `<title>`-tag containing the given text
+- **TestId**: queries by the `data-testid`-attribute. You can setup a different attribute via `configure({testIdAttribute: 'data-my-test-attribute'})`
+
+For more information, check the [testing-library documentation](https://testing-library.com/docs/queries/about).
+
+#### Testing in context
+
+The component might be reliant on some context. In order to wrap it, use the `wrapper`-option. 
+
+```tsx title="Context.test.tsx"
+import { test, expect } from "vitest"
+import { render } from "@solidjs/testing-library"
+import { DataContext, DataConsumer } from "./Data"
+
+const wrapper = (props) => <DataContext value="test" {...props} />
+
+test("receives data from context", () => {
+  const { getByText } = render(() => <DataConsumer />, { wrapper })
+  expect(getByText("test")).toBeInTheDocument()
+});
+```
+
+Wrappers can be re-used if they are created externally. For wrappers with different values, a higher-order component creating the required wrappers can make the tests more concise:
+
+```tsx
+const createWrapper = (value) => (props) => 
+  <DataContext value={value} {...props}/>
+```
+
+<Callout title="Using multiple providers">
+If using multiple providers, [solid-primitives](https://primitives.solidjs.community/) has [`<MultiProvider>`](https://primitives.solidjs.community/package/context#multiprovider) to avoid nesting multiple levels of providers
+</Callout>
+
+#### Testing routes
+
+For convenience, `render` supports the `location`-option that wraps the rendered component in a router pointing at the given location. Since the `<Router>` component is lazily loaded, the first query after rendering needs to be asynchronous, i.e. `findBy...`:
+
+```tsx
+const { findByText } = render(
+  () => <Route path="/article/:id" component={Article} />,
+  { location: "/article/12345" }
+);
+expect(await findByText("Article 12345")).toBeInTheDocument()
+```
+
+### Interacting with components
```

Comment:
This would break the previous enumeration of "render / interact / assert", which should all be headlines level 3. Interaction is not a sub-topic of rendering.